src/libmongoc/tests/test-mongoc-sdam.c

```diff
@@ -443,7 +695,138 @@ test_existing_behavior (void *ctx)
    bson_free (uri_str);
    bson_free (replset_name);
    bson_free (host_and_port);
+}
+
+typedef struct {
+   uint32_t n_heartbeat_succeeded;
+} prose_test_ctx_t;
 
+static void
+heartbeat_succeeded (const mongoc_apm_server_heartbeat_succeeded_t *event)
+{
+   prose_test_ctx_t *ctx;
+
+   ctx =
+      (prose_test_ctx_t *) mongoc_apm_server_heartbeat_succeeded_get_context (
+         event);
+   ctx->n_heartbeat_succeeded++;
+/* The reported duration may be 0 on Windows due to poor clock resolution.
+ * bson_get_monotonic_time () uses GetTickCount64. MS docs say:
+ * "GetTickCount64 function is limited to the resolution of the system timer,
+ * which is typically in the range of 10 milliseconds to 16 milliseconds"
+ */
+#ifndef _WIN32
+   BSON_ASSERT (mongoc_apm_server_heartbeat_succeeded_get_duration (event) > 0);
+#endif
+}
+
+static void
+test_prose_rtt (void *unused)
+{
+   /* Since this tests RTT tracking in the streaming protocol, this test
+    * requires a client pool. */
+   mongoc_client_pool_t *pool;
+   mongoc_uri_t *uri;
+   mongoc_client_t *client;
+   mongoc_collection_t *coll;
+   bson_error_t error;
+   const bson_t *doc;
+   mongoc_cursor_t *cursor;
+   mongoc_apm_callbacks_t *callbacks;
+   prose_test_ctx_t ctx;
+   bson_t cmd;
+   bool ret;
+   int64_t start_us;
+   bool satisfied;
+   int64_t rtt = 0;
+
+   uri = test_framework_get_uri ();
+   mongoc_uri_set_option_as_utf8 (uri, MONGOC_URI_APPNAME, "streamingRttTest");
+   mongoc_uri_set_option_as_int32 (uri, MONGOC_URI_HEARTBEATFREQUENCYMS, 500);
+
+   callbacks = mongoc_apm_callbacks_new ();
+   mongoc_apm_set_server_heartbeat_succeeded_cb (callbacks,
+                                                 heartbeat_succeeded);
+   pool = mongoc_client_pool_new (uri);
+   test_framework_set_pool_ssl_opts (pool);
+   memset (&ctx, 0, sizeof (prose_test_ctx_t));
+   mongoc_client_pool_set_apm_callbacks (pool, callbacks, &ctx);
+   client = mongoc_client_pool_pop (pool);
+
+   /* Run a find command for the server to be discovered. */
+   coll = get_test_collection (client, "streamingRttTest");
+   cursor = mongoc_collection_find_with_opts (
+      coll, tmp_bson ("{}"), NULL /* opts */, NULL /* read prefs */);
+   mongoc_cursor_next (cursor, &doc);
+
+   /* Sleep for two seconds to allow multiple heartbeats to succeed. */
+   _mongoc_usleep (2 * 1000 * 1000);
+
+   /* Set a failpoint to make isMaster commands take longer. */
```

Comment:
[RTT Tests](https://github.com/mongodb/specifications/tree/master/source/server-discovery-and-monitoring/tests#rtt-tests) has a step to "Assert that each ServerDescriptionChangedEvent includes a non-zero RTT." between the 2-second sleep and `configureFailPoint` call. Is that missing?

Reply:
Found it. Thanks!